### DISCUSSION TITLE
How to fetch data on client

### DISCUSSION THREAD

**Discussion by Denllay** [vikejs/vike#1509] [Discussion]
I want to fetch important data on server side using onBeforeRender hook (like authenticated user session). 
But I want to fetch other data on client, showing loading status while fetching on client.

Do i can use onHydrationEnd hook for fetching data on client? I tried this one but getting "content miss match error".

**Comment by brillout** [comment]
> fetch other data on client, showing loading status while fetching on client.

I presume you don't need Vike for that, for example, using a stateful component with `fetch()`, or a library like Telefunc or React Query.
